codeface_extraction/extractions.py

```diff
@@ -292,18 +297,22 @@ def __init__(self, dbm, conf, resdir, csv_writer):
                     AND p.analysisMethod = '{tagging}'
                     AND cd.file IS NOT NULL
 
-                    ORDER BY cd.file, c.id, c.commitHash, cd.entityId
+                    ORDER BY cd.file, cd.entityId, c.id, c.commitHash
                 """
 
     def _reduce_result(self, result):
+        """
+        Removes control characters such as \r\n \x1b \ufffd from string impl and returns a unicode
+        string where all control characters have been replaced by a space
+        """
+
         newResult = []
-        for (commitId, commitHash, changedFiles, fileId, entityId, impl) in result:
-            newImpl = impl.replace("\n", " ")
-            newImpl = newImpl.replace("\r", " ")
-            newImpl = newImpl.replace("\t", " ")
-            newImpl = newImpl.replace(u"\ufffd", " ")
+        for (commitId, commitHash, fileId, entityId, impl) in result:
+            newImpl = impl.encode('utf-8')
+            newImpl = re.sub(r'\\ufff.', ' ', newImpl)
+            newImpl = "".join(ch if unicodedata.category(ch)[0]!="C" else " " for ch in newImpl.decode('unicode-escape'))
```

Comment:
Please make quotation marks and spaces around operators more consistent with other parts of the code and insert some small documentation as suggested in [my earlier comment](https://github.com/se-passau/codeface-extraction/pull/14/files/efea611320c7382f0718ab2ec0371a7ceb5b9158..4c92249431392c6a0bde3b9b1e6b902147c31855#r201637899).

Reply:
@nischikata, can you help me to write a small documentation?